src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,34 +39,43 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String [] addressArray = address.split(",");
+        this.addressBlock = new Block(addressArray[BLOCK_INDEX]);
+        this.addressStreet = new Street(addressArray[STREET_INDEX]);
+        this.addressUnit = new Unit(addressArray[UNIT_INDEX]);
+        this.addressPostalCode = new PostalCode(addressArray[POSTAL_CODE_INDEX]);
         this.value = address;
     }
 
     /**
      * Returns true if a given string is a valid person email.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        boolean isValid = test.matches(ADDRESS_VALIDATION_REGEX);
+        String [] testArray = test.split(", ");
+        isValid = isValid && testArray.length == ADDRESS_ARRAY_SIZE;
+        return isValid;
     }
```

Comment:
Instead of writing like line 56 and 57, should be better if you put them in 1 line, `return isValid && testArray.length == ADDRESS_ARRAY_SIZE;`. You need to also change the header comment, since it isn't checking person email.